src/ngx-gallery-helper.service.ts

```diff
@@ -45,4 +45,24 @@ export class NgxGalleryHelperService {
     private removeSwipeHandlers(id: string): void {
         this.swipeHandlers.delete(id);
     }
+
+    getFileType (fileSource: string): string {
+        if (fileSource.startsWith('data:')) {
+          return fileSource.substr(5, Math.min(fileSource.indexOf(';'), fileSource.indexOf('/')) - 5);
+        }
+        let fileExtension = fileSource.split('.').pop().toLowerCase();
+        if (!fileExtension
```

Comment:
Maybe using a indexOf with an array of file type could be better?

Reply:
Yes `indexOf` is better for most of the cases. But `substr` I think that here it will make this piece of code compatible with any content type. If this component will support other types later. This line can stay the as-is.
However, there was unnecessary usage of `substr` inside the next `else if`. I deleted it in the second commit.